server/src/main/java/io/crate/execution/engine/collect/files/FileReadingIterator.java

```diff
@@ -256,16 +261,20 @@ private String getLine(BufferedReader reader, long startFrom, int retry) throws
                 }
                 break;
             }
-        } catch (SocketTimeoutException e) {
-            if (retry > MAX_SOCKET_TIMEOUT_RETRIES) {
-                URI uri = currentInput.uri();
-                LOGGER.error("Timeout during COPY FROM '" + uri.toString() + "' after " + retry + " retries", e);
-                throw e;
-            } else {
+        } catch (SocketException | SocketTimeoutException e) {
+            if (backoffPolicy.hasNext()) {
+                try {
+                    Thread.sleep(backoffPolicy.next().getMillis());
```

Comment:
Could we use a scheduler instead to not block this thread?

Reply:
Sorry I am confused how to use a scheduler within a `BatchIterator`. Could we have quick chat about it?
Also, how about we let `BatchIteratorBackpressureExecutor` to account for delays from `FileReadingIterator.moveNext()`?
https://github.com/crate/crate/blob/ac60750e1e651b563f5ed14a94c97978ae860287/server/src/main/java/io/crate/execution/engine/indexing/BatchIteratorBackpressureExecutor.java#L169

Reply:
Could you have another look at this draft? @seut